### ISSUE TITLE
agsXMPP dependency no longer available via NuGet

### ISSUE THREAD

**Issue by warwickmm** [duplicati/duplicati#4189]
The agsXMPP package appears to have been removed from NuGet:

https://www.nuget.org/packages/agsXMPP/1.1.1

This is causing our builds to fail.  For example,

https://ci.appveyor.com/project/kenkendk/duplicati/builds/32609331
https://travis-ci.org/github/duplicati/duplicati/builds/682470806?utm_source=github_status&utm_medium=notification

```
WARNING: Unable to find version '1.1.1' of package 'agsXMPP'.
  C:\Users\appveyor\.nuget\packages\: Package 'agsXMPP.1.1.1' is not found on source 'C:\Users\appveyor\.nuget\packages\'.
  C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\: Package 'agsXMPP.1.1.1' is not found on source 'C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\'.
  https://api.nuget.org/v3/index.json: Package 'agsXMPP.1.1.1' is not found on source 'https://api.nuget.org/v3/index.json'.
Errors in packages.config projects
    Unable to find version '1.1.1' of package 'agsXMPP'.
      C:\Users\appveyor\.nuget\packages\: Package 'agsXMPP.1.1.1' is not found on source 'C:\Users\appveyor\.nuget\packages\'.
      C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\: Package 'agsXMPP.1.1.1' is not found on source 'C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\'.
      https://api.nuget.org/v3/index.json: Package 'agsXMPP.1.1.1' is not found on source 'https://api.nuget.org/v3/index.json'.
NuGet Config files used:
    C:\projects\duplicati\NuGet.Config
```

**Comment by kenkendk** [comment]
The s22.xmpp library is also really dated (last change 7 years ago).

There is a copy of the agxmpp (only 6 years old):
https://github.com/loamen/agsXmpp

Maybe we are better off, using that version or perhaps removing the xmpp support?
If we remove it, it would be possible to replicate using a commandline tool, such as [sendxmpp](https://askubuntu.com/questions/533527/how-can-i-send-an-xmpp-message-using-the-command-line) which is at least present on Linux systems.

Other solutions (seems to be updated):
- https://www.nuget.org/packages/Waher.Networking.XMPP/
- https://www.nuget.org/packages/Artalk.Xmpp/

**Comment by kenkendk** [comment]
I made a quick-fix that bundles a custom build of agsXMPP:
https://github.com/duplicati/duplicati/commit/e240b4b267d37a292689979881e19ba865ae50b0
Not a great solution, but it makes the CI builds working at least.

**Comment by warwickmm** [comment]
> I made a quick-fix that bundles a custom build of agsXMPP

This may be the simplest fix, and will buy us time to consider a better solution.

**Comment by warwickmm** [comment]
This was resolved by e240b4b267d37a292689979881e19ba865ae50b0.
